velox/dwio/dwrf/writer/Writer.cpp

```diff
@@ -805,6 +805,18 @@ dwrf::WriterOptions getDwrfOptions(const dwio::common::WriterOptions& options) {
   dwrfOptions.memoryPool = options.memoryPool;
   dwrfOptions.spillConfig = options.spillConfig;
   dwrfOptions.nonReclaimableSection = options.nonReclaimableSection;
+
+  if (options.flushPolicyFactory) {
```

Comment:
Can consolidate dwrf flush policy interface into dwio common flush policy interface? So we don't need this handling? If parquet doesn't need shouldFlushDictionary, then the actual flush policy implementation for parquet could leave it a default implementation? @HuamengJiang Thanks!

Reply:
> Can we consolidate dwrf flush policy interface into dwio common flush policy interface? So we don't need this handling? If parquet doesn't need shouldFlushDictionary, then the actual flush policy implementation for parquet could leave it a default implementation? @HuamengJiang Thanks!

One concern with merging is that having a special case for dictionary encoding is a DWRF implementation detail. Maybe it also fits parquet but it wouldn't fit nimble. 

This was originally desirable for coordinating flushes at memory pressure scenarios across file formats, but that is now taken care of by memory arbitration. So in a sense, we no longer need a common flush policy interface, and we can just create different classes branching on format in the eval engines.

Reply:
> > Can we consolidate dwrf flush policy interface into dwio common flush policy interface? So we don't need this handling? If parquet doesn't need shouldFlushDictionary, then the actual flush policy implementation for parquet could leave it a default implementation? @HuamengJiang Thanks!
> 
> One concern with merging is that having a special case for dictionary encoding is a DWRF implementation detail. Maybe it also fits parquet but it wouldn't fit nimble.
> 
> This was originally desirable for coordinating flushes at memory pressure scenarios across file formats, but that is now taken care of by memory arbitration. So in a sense, we no longer need a common flush policy interface, and we can just create different classes branching on format in the eval engines.

If that's the case, we can just refactor dwrf writer to do cast internally so all the factories we use return a unique dwio io flush policy interface?

@JkSelf do you still need this change? Thanks!

Reply:
Chat offline with @HuamengJiang. He will first make a change to let dwrf writer to use flush factory which return a unique_ptr of dwio flush policy which will simplify this change.

Reply:
@xiaoxmeng Can we move this optimization in the following PRs and merge this PR firstly if no further comment? Thanks.

Reply:
@xiaoxmeng When we convert the base class common::WriterOptions to the subclass dwrf::WriterOptions [here](https://github.com/facebookincubator/velox/pull/8864/files#diff-c00f448770f041f78addaa96e1ef6954732751b57b001950e92865a2f3d830e2R855), we can directly assign flushPolicy to dwrfWriteOptions, so the conversion here is no longer needed.